### ISSUE TITLE
Gatsby Cloud builds & previews are not working

### ISSUE THREAD

**Issue by pepijn-vanvlaanderen** [relate-app/gatsby-source-strapi#18]
When I run a Gatsby Cloud build or Preview build it does not update the build. The preview logs show also state:
```
info There are no new or changed html files to build.
```

**Comment by t-kietzmann** [comment]
I had the same issue, temporary fix for me is to run a complete build (runs of course a bit longer!)

Add this Header to the Webhook in the Strapi Config:
key: x-gatsby-cache
value: false

**Comment by timbrandin** [comment]
I'm working on this, and it seems at first glance being an issue with Strapi webhooks. I'm actually considering building a Strapi plugin specifically for Gatsby with it's own webhook that calls it correctly and makes the build work as it should, but also that gives us the preview button for using the preview function.

**Comment by pepijn-vanvlaanderen** [comment]
How should the build/preview webhooks look like? According to [Gasby docs](https://support.gatsbyjs.com/hc/en-us/articles/360052324394-Build-and-Preview-Webhooks), I think you only need to pass a header:
```
"x-gatsby-cloud-data-source": "gatsby-source-strapi"
```
This does not seem to work unfortunately.

**Comment by timbrandin** [comment]
I have a prototype working now, will open source it soon, but the gist is sending a POST request to the webhook url without any body worked the best. This plugin already is keeping track of what changed through the grapqhl api and a date saved in gatsby cloud cache key, so the incremental builds and previews take about ~5-15 sec on a non cold start.

Lookout for a package named: @relate-app/strapi-gatsby-cloud or something 👯

**Comment by timbrandin** [comment]
> ```
> "x-gatsby-cloud-data-source": "gatsby-source-strapi"
> ```

Will test this on the new strapi plugin.

**Comment by timbrandin** [comment]
Fixed in `4.4.0` and with this plugin it should work: https://github.com/relate-app/strapi-gatsby-cloud
